### DISCUSSION TITLE
Живой поиск тэгов

### DISCUSSION THREAD

**Discussion by klimashkin** [PastVu/pastvu#11] [Ideas / Идеи и предложения]
Сейчас на одну Москву приходится уже больше 8 тысяч тэгов, 80% которых повторения или близкие по значению.
Такое разрастание количества тэгов негативно влияет на производительность базы, особенно в связи с расширением.
Для снятия проблемы нужна подсказка по существующим тэгам, когда пользователь начинает вводить новый тэг в поле.
Например, как это сделано в FB и G+

**Comment by Duche3d** [comment]
очень важен инструмент их редактирования и корректного объединения, потому что большая часть их этих 8000 это дубликаты с разным написанием, просто с орфографическими ошибками, или вообще некорректно придуманные

Dec 28, 2012, в 9:11 PM, klimashkin notifications@github.com написал(а):

**Comment by klimashkin** [comment]
Да, будет полезная штука - администрирование тэгов. Пока не очень представляю себе это в интерфейсе. Возможно нужно будет выделить в отдельный кейс

**Comment by Duche3d** [comment]
ну это в админском интерфейсе должно быть

Dec 28, 2012, в 10:16 PM, klimashkin notifications@github.com написал(а):

**Comment by klimashkin** [comment]
Да. Надо подумать как в интерфейсе сделать объединение из 8000 строк. Глазами сложно искать по такому списку

**Comment by Duche3d** [comment]
ну я думаю нужно
1. иметь возможность вывести все одним списком все или поиском по сочетанию букв
2. иметь чекбоксы напротив тегов которые хочешь объединить и кнопку - объединить/редактировать, нажатием на которую и будет происходить корректная замена выбранных тегов
8000 это вначале трудно будет разобрать, но постепенно причешется
3. далее можно у тегов сделать статус "новый" - это теги который пользователи добавили, а не выбрали из списка уже имеющихся и соответственно отдельная кнопка для модераторов для просмотра и редактирования новых тегов, чтоб отслеживать состояние и недопускать разростание
4. и очень важное это поиск и уничтожение или объединение тегов которые назначены только одному фото - такие теги бессмысленны, поскольку инструмент тегов это в первую очередь инструмент создания подборок схожих фото, а не уникальная их идентификация (ну вообще в идеале в писке тегов нужно иметь информацию о количестве фото с соттветсвующим тегом)
5. Облако или список тегов, поиск по тегам - одним словом инструмент для просмотра сайта через теги. Сейчас это сделано довольно странно - открываешь какое-то фото, видишь у него тэг и тыкнув на него можешь увидеть другие фото с тем же тегом. Посмотреть какие теги есть или поискать по ним невозможно.

**Comment by paul-k-pastvu** [comment]
Общие контуры задачи обсуждаются в #20
Живой поиск (поисковые подсказки = search suggestions) сделан для выбора регионов (но требует доработки). и ровно такой же можно будет приделать  к тегам
